### PR TITLE
test: ensure ipv6 compatibility for cli and config

### PR DESCRIPTION
**Pull request by ddebko**
# Summary

This PR ensures that the CLI commands inputed by the user can include field values that contain IPv6 addresses. This PR also ensures that the config file may contain field values that have IPv6 addresses.

### CODE REVIEW

File: internal/cmd/config/config.go [hunk 1]
```diff
@@ -1278,6 +1285,16 @@ func (c *Config) SetupControllerPublicClusterAddress(flagValue string) error {
...
+	// net.JoinHostPort() will add a second pair of square brackets.
+	host = squareBrackets.ReplaceAllString(host, "")
+	if host == "" {
+		// return early if a empty host was provided
+		c.Controller.PublicClusterAddr = net.JoinHostPort(host, port)
+		return nil
+	}
+
 	c.Controller.PublicClusterAddr = net.JoinHostPort(host, port)
 	return nil
```
Comment (kheina) [inline internal/cmd/config/config.go:1299]: I am curious what the purpose is of this return early clause if we return immediately afterwards

**Review by kheina — APPROVED** [review]
LGTM, just had one question

File: internal/cmd/config/config.go (hunk 1 above)
Comment (ddebko) [reply internal/cmd/config/config.go:1299]: Thanks for catching this. My original implementation parsed the host value to check if it was a valid IPv4 or IPv6 address, but I had to remove this logic because it would break some unix listener and enos tests. This if conditional was an edge case to ensure that these tests passed as expected, but now I can remove this logic.

**Review by kheina — APPROVED** [review]
LGTM
